platform-sdk/swirlds-common/src/main/java/com/swirlds/common/scratchpad/internal/StandardScratchpad.java

```diff
@@ -0,0 +1,375 @@
+    private final AutoClosableLock lock = Locks.createAutoLock();
+    private final Path scratchpadDirectory;
+    private long nextScratchpadIndex;
+
+    private final int fileVersion = 1;
+
+    /**
+     * Create a new scratchpad.
+     *
+     * @param platformContext the platform context
+     * @param selfId          the ID of this node
+     * @param clazz           the enum class that defines the scratchpad fields
+     * @param id              the unique ID of this scratchpad (creating multiple scratchpad instances on the same node
+     *                        with the same unique ID has undefined (and possibly undesirable) behavior. Must not
+     *                        contain any non-alphanumeric characters, with the exception of the following characters:
+     *                        "_", "-", and ".". Must not be empty.
+     */
+    public StandardScratchpad(
+            @NonNull final PlatformContext platformContext,
+            @NonNull final NodeId selfId,
+            @NonNull final Class<K> clazz,
+            @NonNull final String id) {
+
+        final StateConfig stateConfig = platformContext.getConfiguration().getConfigData(StateConfig.class);
+        scratchpadDirectory = stateConfig
+                .savedStateDirectory()
+                .resolve(SCRATCHPAD_DIRECTORY_NAME)
+                .resolve(Long.toString(selfId.id()))
+                .resolve(id);
+
+        if (id.isEmpty()) {
+            throw new IllegalArgumentException("scratchpad ID must not be empty");
+        }
+        if (!id.matches("[a-zA-Z0-9_.-]+")) {
+            throw new IllegalArgumentException(
+                    "scratchpad ID must only contain alphanumeric characters, '_', '-', " + "and '.'");
+        }
+
+        this.id = id;
+        fields = EnumSet.allOf(clazz);
+
+        indexToFieldMap = new HashMap<>();
+        for (final K key : fields) {
+            final K previous = indexToFieldMap.put(key.getFieldId(), key);
+            if (previous != null) {
+                throw new RuntimeException("duplicate scratchpad field ID: " + key.getFieldId());
+            }
+        }
+
+        loadFromDisk();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void logContents() {
+        final TextTable table = new TextTable().setBordersEnabled(false);
+
+        for (final K field : fields) {
+            final SelfSerializable value = data.get(field);
+            if (value == null) {
+                table.addToRow(field.name(), "null");
+            } else {
+                table.addRow(field.name(), value.toString());
+            }
+        }
+
+        logger.info(
+                STARTUP.getMarker(),
+                """
+                        Scratchpad {} contents:
+                        {}""",
+                id,
+                table.render());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public <V extends SelfSerializable> V get(@NonNull final K key) {
```

Comment:
method should have `@Nullable` annotation

Reply:
added